elk/training/train.py

```diff
@@ -78,88 +78,129 @@ def train_reporter(
         layer: int,
         devices: list[str],
         world_size: int = 1,
-    ) -> pd.Series:
+    ) -> pd.DataFrame:
         """Train a single reporter on a single layer."""
         self.make_reproducible(seed=self.cfg.net.seed + layer)
-
         device = self.get_device(devices, world_size)
 
-        x0, x1, val_x0, val_x1, train_gt, val_gt, val_lm_preds = self.prepare_data(
-            device, layer
-        )
-        pseudo_auroc = self.get_pseudo_auroc(layer, x0, x1, val_x0, val_x1)
+        train_dict = self.prepare_data(device, layer, "train")
+        val_dict = self.prepare_data(device, layer, "val")
+
+        # Can't figure out a way to make this line less ugly
+        hidden_size = next(iter(train_dict.values()))[0].shape[-1]
+        reporter_dir, lr_dir = self.create_models_dir(assert_type(Path, self.out_dir))
+        pseudo_clf = self.get_pseudo_classifier(train_dict, device)
 
         if isinstance(self.cfg.net, CcsReporterConfig):
-            reporter = CcsReporter(x0.shape[-1], self.cfg.net, device=device)
+            assert len(train_dict) == 1, "CCS only supports single-task training"
```

Comment:
While this isn't high priority, we could be a little more lenient than this and allow training on mixtures of datasets when the shapes of the tensors are the same (same num_variants, num_choices).